### ISSUE TITLE
Request for Code Release Plan

### ISSUE THREAD

**Issue by robertzzq** [DevikalyanDas/npgs#2]
Hello,

Thank you for the excellent work! Is there any plan to release the code (or parts of the code)?

Thank you in advance, and I look forward to hearing from you!

Regards,
Robert

**Comment by DevikalyanDas** [comment]
Hello Robert,

I am sorry for the late reply. I am busy with some other deadlines and not able to give time for the code release. I will release the entire code that was used for the paper results but it can be delayed like in the month of December. I apologise for this. Thanks for your interest in our work.  Here, is the snippet for the flow loss.
Best,
Devikalyan Das

## Flow Loss
def safe_sqrt(A, eps_1=float(1e-6)):
    """
    performs safe differentiable sqrt
    """

    return (torch.clamp(A,min=float(0)) + eps_1).sqrt()

def soft_huber(dfsq, eps):
    loss = eps * (safe_sqrt(1.0 + dfsq)-1.0)
    return loss

class Flow_Consistency_Loss(nn.Module):
    def __init__(self, epsilon=0.01):
        super(Flow_Consistency_Loss,self).__init__()
        self.epsilon = float(epsilon)

    def forward(self,rendered_1,flow_wrapped_1, mask_1):
        x,y,mask_i = rendered_1,flow_wrapped_1,mask_1[...,None]

        diff = (x * mask_i) - (y * mask_i)

        bs,h,w,ch = x.shape

        diff = diff.reshape((bs,-1))
        mask_sum = mask_i.reshape((bs,-1))
        mask_sum = torch.clamp(mask_sum.sum(1), 1.)

        dist_hub =  soft_huber((diff**2)/(self.epsilon**2),eps=self.epsilon)
        
        fin_val = (dist_hub.sum(1)/mask_sum).mean()
        
        return fin_val

**Comment by robertzzq** [comment]
Thank you very much!

**Comment by luminousking** [comment]
Hi,

Thx for your release for stage1 code! Do you have an ETA of stage2 code? 

Many thanks,
